### PR TITLE
Added Endpoint Enrichment - Generic v2.1 playbook + v2.1 test playbook,  Endpoint Enrichment - Cylance Protect v2

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
In progress - need documentation

## Related Issues
Related to: https://github.com/demisto/etc/issues/16881

## Description
Added a `Endpoint Enrichment - Cylance Protect v2` playbook, to enrich an endpoint using the Cylance Protect v2 integration.

Added the `Endpoint Address Enrichment - Generic v2.1` playbook. The playbook features the following improvements over the previous one:
- Changed enrichment using AD to use AD v2
- Changed enrichment using Carbon Black Enterprise Response to use Carbon Black Enterprise Response v2
- Changed enrichment using Cylance Protect to use Cylance Protect v2

Added a test playbook `Endpoint Enrichment - Generic v2.1 - Test`.

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Playbooks/playbook-Endpoint_Enrichment_-_Cylance_Protect_v2.yml [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    ignoreworker: false
+  "1":
+    id: "1"
+    taskid: 50264655-eac1-4439-809b-32f93a19c825
+    type: condition
+    task:
+      id: 50264655-eac1-4439-809b-32f93a19c825
+      version: -1
+      name: Is Cylance Protect v2 enabled?
+      description: Checks if there is any active instance of the Cylance Protect v2
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Cylance_Protect_v2.yml]: Checks if there is an active instance of the Cylance Protect v2 integration enabled.

File: Playbooks/playbook-Endpoint_Enrichment_-_Cylance_Protect_v2.yml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    ignoreworker: false
+  "3":
+    id: "3"
+    taskid: ec7a9e06-60d0-4c25-8ed4-1cb9d0fe92f6
+    type: regular
+    task:
+      id: ec7a9e06-60d0-4c25-8ed4-1cb9d0fe92f6
+      version: -1
+      name: Set enriched device to context
+      description: Sets the device that has been enriched into context.
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Cylance_Protect_v2.yml]: Sets the device that was enriched into context.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 3]
```diff
@@ -0,0 +1,702 @@
+id: Endpoint Enrichment - Generic v2.1
+version: -1
+fromversion: 4.1.0
+name: Endpoint Enrichment - Generic v2.1
+description: |-
+  Enrich an endpoint by hostname using one or more integrations.
+  Currently, the following integrations are supported:
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Supported integrations:

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+    ignoreworker: false
+  "2":
+    id: "2"
+    taskid: c9799048-288a-40cc-8d2f-c561d0ea2f6b
+    type: condition
+    task:
+      id: c9799048-288a-40cc-8d2f-c561d0ea2f6b
+      version: -1
+      name: Is McAfee ePolicy Orchestrator enabled?
+      description: Checks if there is any active instance of the McAfee ePolicy Orchestrator
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Checks if there is an active instance of the McAfee ePolicy Orchestrator integration enabled.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+    timertriggers: []
+    ignoreworker: false
+  "5":
+    id: "5"
+    taskid: 44a06d71-1772-47af-8d41-8c6dbdd149e7
+    type: regular
+    task:
+      id: 44a06d71-1772-47af-8d41-8c6dbdd149e7
+      version: -1
+      name: Get host information from ePO
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Get host information from McAfee ePO

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 6]
```diff
@@ -0,0 +1,702 @@
...
+    ignoreworker: false
+  "8":
+    id: "8"
+    taskid: 9fa921fa-d196-40ba-8419-ed0c4f838ab8
+    type: condition
+    task:
+      id: 9fa921fa-d196-40ba-8419-ed0c4f838ab8
+      version: -1
+      name: Is Carbon Black Enterprise Response enabled?
+      description: Checks if there is any active instance of the Carbon Black Enterprise
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Checks if there is an active instance of the Carbon Black Enterprise Response integration enabled.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 7]
```diff
@@ -0,0 +1,702 @@
...
+    ignoreworker: false
+  "10":
+    id: "10"
+    taskid: 216aed2b-2f34-4c5e-8c54-60b978218896
+    type: condition
+    task:
+      id: 216aed2b-2f34-4c5e-8c54-60b978218896
+      version: -1
+      name: Is CrowdStrike Falcon Host enabled?
+      description: Checks if there is any active instance of the CrowdStrike Falcon
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Checks if there is an active instance of the CrowdStrike Falcon Host integration enabled.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 8]
```diff
@@ -0,0 +1,702 @@
...
+    ignoreworker: false
+  "22":
+    id: "22"
+    taskid: 6a8160b7-fe32-4ab0-8396-e9bcafc98194
+    type: condition
+    task:
+      id: 6a8160b7-fe32-4ab0-8396-e9bcafc98194
+      version: -1
+      name: Is Active Directory Query v2 enabled?
+      description: Checks if there is any active instance of the Active Directory
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Checks if there is an active instance of the Active Directory Query v2 integration enabled.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 9]
```diff
@@ -0,0 +1,702 @@
...
+    id: "23"
+    taskid: 23c75906-3b4e-42e8-8e32-c883d5f06fb3
+    type: regular
+    task:
+      id: 23c75906-3b4e-42e8-8e32-c883d5f06fb3
+      version: -1
+      name: Get host information from Active Directory
+      description: |-
+        Uses Active Directory to retrieve detailed information about the computer of the account specified. The computer can be specified by name, email address, or as an Active Directory Distinguished Name (DN).
+        If no filters are specified, the result will show all computers.
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: If no filters are specified, returns all computers.

File: Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml [hunk 10]
```diff
@@ -0,0 +1,702 @@
...
+    ignoreworker: false
+  "25":
+    id: "25"
+    taskid: 71baffa3-cd35-4cb0-8f50-0bd4c45fbc70
+    type: playbook
+    task:
+      id: 71baffa3-cd35-4cb0-8f50-0bd4c45fbc70
+      version: -1
+      name: CrowdStrike Endpoint Enrichment
+      description: Enrich Endpoint with CrowdStrike
```
Comment (kirbles19) [inline Playbooks/playbook-Endpoint_Enrichment_-_Generic_v2.1.yml]: Enrich an endpoint using CrowdStrike.

**Comment by idovandijk** [comment]
@michalgold - done.
@kirbles19 - can you review the changes? Look for the new descriptions under "Set device list" and under "Set enriched device details" in the Cylance Protect v2 YML file.

**Pull request merged**